### DISCUSSION TITLE
Reading XLS is ignoring DecimalSeparator settings

### DISCUSSION THREAD

**Discussion by cyklop** [PHPOffice/PhpSpreadsheet#2345] [Q&A] [answered]
Hi,
I'm reading xls files and trying to set the decimal separator by `PhpOffice\PhpSpreadsheet\Shared\StringHelper::setDecimalSeparator(',');`
Seems like this setting is only for writing data. 
`$spreadsheet->getActiveSheet()->toArray()`
function is changing values to decimalSeparator `.`

is there an other way to get values as they are showing up in the excel files localy

**Comment by oleibman** [comment, marked as answer]
I believe that Excel stores all its data using period for decimal separator - see for example https://social.technet.microsoft.com/Forums/office/en-US/eaa4c7f6-197a-4b33-bc5f-20896e5a7e3a/workbook-or-worksheet-specific-decimal-separator?forum=excel. There is a global option which allows you to change the decimal separator when data is displayed, but it is applied globally, not to specific workbooks. So, Excel doesn't actually store the decimal separator in use with the workbook; thus, there is no way for for PhpSpreadsheet to automatically do as you wish. This is especially true if you use the default Number Format `General`. If you set the style to use a custom Number Format, you might be able to get the result you are looking for with the `getFormattedValue` function; you would, of course, have to iterate through all the cells individually rather than use `toArray`. As an alternative, you might be able to use the PHP function `number_format` on each of your values.
